plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -0,0 +1,331 @@
+        this.formatVersion = requireNonNull(icebergConfig, "icebergConfig is null").getFormatVersion();
+        this.isUsingSystemSecurity = requireNonNull(securityConfig, "securityConfig is null").getSecuritySystem() == SYSTEM;
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "migrate",
+                ImmutableList.of(
+                        new Procedure.Argument("SCHEMA", VARCHAR),
+                        new Procedure.Argument("TABLE", VARCHAR)),
+                MIGRATE.bindTo(this));
+    }
+
+    public void migrate(ConnectorSession session, String schemaName, String tableName)
+    {
+        // this line guarantees that classLoader that we stored in the field will be used inside try/catch
+        // as we captured reference to PluginClassLoader during initialization of this class
+        // we can use it now to correctly execute the procedure
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doMigrate(session, schemaName, tableName);
+        }
+    }
+
+    public void doMigrate(ConnectorSession session, String schemaName, String tableName)
+    {
+        SchemaTableName sourceTableName = new SchemaTableName(schemaName, tableName);
+        TrinoCatalog catalog = catalogFactory.create(session.getIdentity());
+        HiveMetastore metastore = metastoreFactory.createMetastore(Optional.of(session.getIdentity()));
+
+        io.trino.plugin.hive.metastore.Table hiveTable = metastore.getTable(schemaName, tableName).orElseThrow(() -> new TableNotFoundException(sourceTableName));
+        String transactionalProperty = hiveTable.getParameters().get(TRANSACTIONAL);
+        if (parseBoolean(transactionalProperty)) {
+            throw new TrinoException(NOT_SUPPORTED, "Migrating transactional tables is unsupported");
+        }
+        if (!hiveTable.getTableType().equalsIgnoreCase(MANAGED_TABLE.name())) {
+            throw new TrinoException(NOT_SUPPORTED, "The procedure supports migrating only managed tables: " + hiveTable.getTableType());
+        }
+        // TODO: Deny Hudi tables. Ideally, this should check the table is exactly Hive table instead of verifying non-Delta Lake or non-Iceberg
+        if (isDeltaLakeTable(hiveTable)) {
+            throw new TrinoException(NOT_SUPPORTED, "The procedure doesn't support migrating Delta Lake tables");
+        }
+        if (isIcebergTable(hiveTable)) {
+            throw new TrinoException(NOT_SUPPORTED, "The table is already an Iceberg table");
+        }
+
+        Schema schema = toIcebergSchema(concat(hiveTable.getDataColumns().stream(), hiveTable.getPartitionColumns().stream()).collect(toImmutableList()));
+        NameMapping nameMapping = MappingUtil.create(schema);
+        HiveStorageFormat storageFormat = extractHiveStorageFormat(hiveTable);
+        verifySupportedStorageFormat(storageFormat);
```

Comment:
The table format doesn't matter if this is partitioned table.
You can skip this.